cachito/workers/pkg_managers/general_js.py

```diff
@@ -451,3 +462,31 @@ def upload_non_registry_dependency(dep_identifier, version_suffix, verify_script
 
         repo_name = get_js_hosted_repo_name()
         nexus.upload_artifact(repo_name, "npm", modified_dep_archive)
+
+
+def _verify_checksum(file_path, checksum_info, chunk_size=10240):
+    """
+    Verify the checksum of the file at the given path matches the expected checksum info.
+
+    :param str file_path: the path to the file to be verified
+    :param ChecksumInfo checksum_info: the expected checksum information
+    :param int chunk_size: the amount of bytes to read at a time
+    :raise CachitoError: if the checksum is not as expected
+    """
+    hasher = hashlib.new(checksum_info.algorithm)
```

Comment:
Since the input to this comes from the package-lock.json file, in theory, the algorithm could be an arbitrary value. I think you should raise a nice error saying that the hash algorithm is not recognized.

Reply:
That's a good point. Wrapped the error.